app/reducers/history.ts

```diff
@@ -0,0 +1,40 @@
+import { initialAppState } from './initialAppState';
+import { makeValidAddressBarUrl } from '$Utils/urlHelpers';
+import { TYPES as TABS_TYPES } from '$Actions/tabs_actions';
+import { logger } from '$Logger';
+
+const initialState = initialAppState.history;
+
+const updateHistory = ( state, payload ) => {
+    if ( payload.url ) {
+        const url = makeValidAddressBarUrl( payload.url );
+        if ( url !== state[state.length - 1] ) {
+            const newState = [...state];
+            newState.push( url );
```

Comment:
Do we only update history if it's not the same as the last one?

What if it's the same as one 5 sites before? Does that _previous_ site remain? Do we still show it if it does? (Or do we only show the latest instance? (I realise this isn't all reducer specific)